### PR TITLE
Improve eligibility conditions

### PR DESCRIPTION
**Pull request by SolalPirelli**
* Add a requirement for participants to live, work or study in Switzerland (and eliminate the rather funny "PolyProg will not help contestants obtain a visa").

* Remove the concepts of a waiting list and of multiple registration deadlines; use a "register and confirm" system instead.

* Add the ability for organizers to exempt teams from registration criteria if absolutely needed.

* Simplify the general wording.

### CODE REVIEW

**Comment by SolalPirelli** [comment]
Some minor improvements courtesy of @tharvik's feedback (via Skype).

He also suggested removing the payment exemptions from the rules and putting them on the website instead. What do others think about that?

**Comment by Rabyss** [comment]
I find this nice and I have nothing to disagree with in the current state of this pull request.

**Comment by zenhaeus** [comment]
Seems fine to me.

File: Rules.md [hunk 1]
```diff
@@ -9,73 +9,62 @@ The Contest environment should allow first and foremost the dissemination of kno
...
+  2.4 **Payment**
+    
+    9. Teams are required to pay registration fees in the amount indicated on the contest's website.
 
-    23. Teams not showing up during the team registration may be disqualified from the contest and forfeit their registration fee and t-shirt. The Contest Judge may make exceptions for contestants living far away from Lausanne, at his discretion.
+    10. At their discretion, contest organizers may allow teams to pay the registration fee on-site when checking in.
+    
+  2.5 **Registration confirmation**
 
-    24. Teams not showing up, sleeping, behaving rudely or interfering in any way during the sponsors' presentations will be disqualified and forfeit their registration fee and t-shirt.
```
Comment (pierluca) [inline Rules.md]: Not sure removing this is a great idea. This has never happened, but I don't see why a defense against such behaviors is worth forfeiting.

File: Rules.md [hunk 2]
```diff
@@ -9,73 +9,62 @@ The Contest environment should allow first and foremost the dissemination of kno
...
+    
+    19. At any time before the registration deadline, teams or team members may withdraw from the contest for any reason and have their registration fee reimbursed.
+    
+  2.7 **Registration fee exemptions**
 
-    29. Teams who miss the third deadline may be put on the waiting list, at the Judge's discretion.
+    20. The registration fee is waived for members of the PolyProg association.
 
-    30. Teams will be removed from the waiting list in the order in which they solve the task provided (1) enough resources are available and (2) the Judge has not yet closed registration. At the Judge's discretion, teams who have not yet payed their registration fees might stay on the waiting list despite there being enough space available.
+    21. VIS and SOI may pay for the registration fee and part of the travel expenses for contestants meeting certain criteria. Contestants should contact the VIS or SOI president for more information.
```
Comment (pierluca) [inline Rules.md]: Is this part of the rules or should this just be a note on the website? I tend to think the latter

Comment (SolalPirelli) [inline Rules.md]: Yeah, IMHO this should be part of the website. I planned to do another pass of changes anyway to remove the things that don't belong in rules, such as the exact hardware/software specs of the machines; that'll be part of it.

**Pull request merged**